src/goto-instrument/aggressive_slicer.h

```diff
@@ -0,0 +1,100 @@
+/*******************************************************************\
+
+Module: Aggressive Slicer
+
+Author: Elizabeth Polgreen, polgreen@amazon.com
+
+\*******************************************************************/
+
+/// \file
+/// Aggressive slicer
+/// Consider the control flow graph of the goto program and a criterion
+/// description of aggressive slicer here
+
+#ifndef CPROVER_GOTO_INSTRUMENT_AGGRESSIVE_SLICER_H
+#define CPROVER_GOTO_INSTRUMENT_AGGRESSIVE_SLICER_H
+
+#include <list>
+#include <string>
+#include <unordered_set>
+
+#include <util/irep.h>
+
+#include <analyses/call_graph.h>
+
+#include <linking/static_lifetime_init.h>
+
+class goto_modelt;
+class message_handlert;
+
+/// \brief The aggressive slicer removes the body of all the functions except
+/// those on the shortest path, those within the call-depth of the
+/// shortest path, those given by name as command line argument,
+/// and those that contain a given irep_idt snippet
+/// If no properties are set by the user, we preserve all functions on
+/// the shortest paths to each property.
+class aggressive_slicert
+{
+public:
+  aggressive_slicert(goto_modelt &_goto_model, message_handlert &_msg)
+    : preserve_all_direct_paths(false),
+      goto_model(_goto_model),
+      message_handler(_msg)
+  {
+    start_function = goto_model.goto_functions.entry_point();
```

Comment:
Not sure why this needs to be stored, and if necessary, make it `const`.

Reply:
it's being used multiple times, so seemed better to store it than to call `goto_model.goto_functions.entry_point()` multiple times